pandas/core/groupby.py

```diff
@@ -2608,6 +2608,8 @@ def _cython_agg_blocks(self, how, numeric_only=True):
         for block in data.blocks:
 
             values = block._try_operate(block.values)
```

Comment:
I'd like to do this differently.

move the ``._try_operate`` call to inse ``.aggregate`` (its line 1533). Then you can use do the ``iNaT`` to ``np.nan`` there